app/views/reservations/new.html.haml

```diff
@@ -25,7 +25,14 @@
 
 = simple_form_for([@order, @order_detail, @reservation]) do |f|
   = f.error_messages
-  = render partial: 'account_field', locals: { f: f } unless @order_detail.bundled?
+
+  - unless @order_detail.bundled?
+    - accounts = @order.user.accounts_for_product(@instrument)
```

Comment:
Even if you were to just leave it in the view, I would lean towards moving it into `_account_field`.